backend/src/main/java/rocks/inspectit/gepard/agentmanager/connection/model/dto/CreateConnectionRequest.java

```diff
@@ -0,0 +1,37 @@
+/* (C) 2024 */
+package rocks.inspectit.gepard.agentmanager.connection.model.dto;
+
+import jakarta.validation.constraints.NotNull;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.UUID;
+import lombok.Builder;
+import rocks.inspectit.gepard.agentmanager.agent.model.Agent;
+import rocks.inspectit.gepard.agentmanager.connection.model.Connection;
+
+/** Represents a connection request from an agent. */
+@Builder
```

Comment:
Does the Builder pattern bring any benefit in this class? 
If every property should not be null, then there basically is just one constructor with every property necessary. Why not just use this constructor? With the Builder we allow "illegal" objects.
Or does it just simplify testing?

Reply:
Good point, will remove it.